unity-game/Assets/Scripts/ElevatorScripts/ManualElevator.cs

```diff
@@ -0,0 +1,226 @@
+            {
+                theEndRightBarrier.SetActive(false);
+            }
+        }
+
+
+
+        if (onEndPoint)
+        {
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == true)
+            {
+                theStartLeftBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == false)
+            {
+                theStartLeftBarrier.SetActive(false);
+            }
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == true)
+            {
+                theStartRightBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == false)
+            {
+                theStartRightBarrier.SetActive(false);
+            }
+        }
+
+
+
+        if (reachedDestination)
+        {
+            readyTimer = .5f;
+            reachedDestination = false;
+        }
+
+
+
+        if (!notMoving)
+        {
+            //readyTimer -= Time.deltaTime;
+            objectToMove.transform.position = Vector3.MoveTowards(objectToMove.transform.position, currentTarget, moveSpeed * Time.deltaTime);
+            readyTimer = 0f;
+            theLeftWall.SetActive(true);
+            theRightWall.SetActive(true);
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == true)
+            {
+                theStartLeftBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == false)
+            {
+                theStartLeftBarrier.SetActive(false);
+            }
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == true)
+            {
+                theStartRightBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == false)
+            {
+                theStartRightBarrier.SetActive(false);
+            }
+            if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endLeftBarrier == true)
+            {
+                theEndLeftBarrier.SetActive(true);
+            }
+            else if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endLeftBarrier == false)
+            {
+                theEndLeftBarrier.SetActive(false);
+            }
+            if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endRightBarrier == true)
+            {
+                theEndRightBarrier.SetActive(true);
+            }
+            else if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endRightBarrier == false)
+            {
+                theEndRightBarrier.SetActive(false);
+            }
+        }
+
+
+
+        if (readyTimer == .5f)
+        {
+            if (Input.GetButtonDown("Interact"))
```

Comment:
this is listening globally, every instance will get triggered whenever the interact input is received even when the player is on the other side of the map. `ElevatorButtonController` should have an interaction distance limit which could then call a `toggleMovement` function in this class